macros/scripts/audio.asm

```diff
@@ -51,51 +81,60 @@ tempo: MACRO
 	bigdw \1 ; tempo
 ENDM
 
-	enum dutycycle_cmd ; $db
-dutycycle: MACRO
-	db dutycycle_cmd
-	db \1 ; duty_cycle
+	enum duty_cycle_cmd ; $db
+duty_cycle: MACRO
+	db duty_cycle_cmd
+	db \1 ; duty cycle
+ENDM
+
+	enum volume_envelope_cmd ; $dc
+volume_envelope: MACRO
+	db volume_envelope_cmd
+	IF \2 < 0
+		db (\1 << 4) | (%1000 | (\2 * -1)) ; volume envelope
+	ELSE
+		db (\1 << 4) | \2 ; volume envelope
+	ENDC
 ENDM
 
-	enum intensity_cmd ; $dc
-intensity: MACRO
-	db intensity_cmd
-	db \1 ; intensity
+	enum pitch_sweep_cmd ; $dd
+pitch_sweep: MACRO
+	db pitch_sweep_cmd
+	IF \2 < 0
+		db (\1 << 4) | (%1000 | (\2 * -1)) ; pitch sweep
+	ELSE
+		db (\1 << 4) | \2 ; pitch sweep
+	ENDC
 ENDM
 
-	enum soundinput_cmd ; $dd
-soundinput: MACRO
-	db soundinput_cmd
-	db \1 ; input
+	enum duty_cycle_pattern_cmd ; $de
+duty_cycle_pattern: MACRO
+	db duty_cycle_pattern_cmd
+	db (\1 << 6) | (\2 << 4) | (\3 << 2) | (\4 << 0) ; duty cycle pattern
 ENDM
 
-	enum sound_duty_cmd ; $de
-sound_duty: MACRO
-	db sound_duty_cmd
-if _NARG == 4
-	db \1 | (\2 << 2) | (\3 << 4) | (\4 << 6) ; duty sequence
-else
-	db \1 ; LEGACY: Support for one-byte duty value
-endc
+	enum toggle_sfx_cmd ; $df
+toggle_sfx: MACRO
+	db toggle_sfx_cmd
 ENDM
 
-	enum togglesfx_cmd ; $df
-togglesfx: MACRO
-	db togglesfx_cmd
+; for compatibility with red
```

Comment:
Unless you can reuse a pokered .asm file with 100% compatibility, I don't think it's worth having macros just for this purpose. Plus, pokered can just as well rename its `execute_music` to `toggle_sfx`. And compatibility macros would make more sense in macros/legacy.asm anyway.

Speaking of which, we have years of .asm files by various artists that ought to stay usable. The pre-PR650 macros should be defined in legacy.asm, or when the names haven't changed, use `_NARG` checks commented as `; LEGACY`.

Reply:
> Unless you can reuse a pokered .asm file with 100% compatibility, I don't think it's worth having macros just for this purpose.

I disagree.

> Plus, pokered can just as well rename its execute_music to toggle_sfx.

In another comment I explain why this is not the case.

> And compatibility macros would make more sense in macros/legacy.asm anyway.

Perhaps. But I think this is misleading because "legacy" macros are deprecated formats from the past. This doesn't really apply to `execute_music` or `toggle_perfect_pitch`. But I see your point. It would be fine if these moved to macros/legacy.asm.

> Speaking of which, we have years of .asm files by various artists that ought to stay usable. The pre-PR650 macros should be defined in legacy.asm, or when the names haven't changed, use _NARG checks commented as ; LEGACY.

Yes, I will work on this.

Reply:
I can live with `drum_note` and `drum_speed`, then. It might even be worth having macros/pokered.asm for everything compatibility-related: we have things like `MOON_STONE_RED` and `red_box_struct` that could be moved.

Reply:
I think things like red items, box structs, and audio macros are better kept in their respective neighborhoods rather than thrown into a macros/pokered.asm bin

Reply:
Box structs actually have a reason to exist here, due to the time capsule.

Reply:
These macros were removed and instead mentioned in docs/music_commands.md